app/controllers/account/settings.js

```diff
@@ -43,5 +65,24 @@ export default Controller.extend({
       this.flashes.clear();
       this.flashes.error('Something went wrong and your email settings were not saved.');
     }
-  }).restartable()
+  }).restartable(),
+
+  setPrivateInsights: task(function* () {
+    let val = this.get('privateInsightsVisibility');
+    try {
+      yield this.preferences.set('private_insights_visibility', val);
+      this.flashes.clear();
+      this.flashes.success(`Your private build insights are now ${val}.`);
+    } catch (err) {
+      this.flashes.clear();
+      this.flashes.error('Something went wrong and your insights settings were not saved.');
+    }
+    this.set('isShowingInsightsVisibilityModal', false);
+  }).restartable(),
```

Comment:
in case you haven’t already started on an acceptance test, I wanted to point out that we’re in the midst of a slow migration to a newer format for acceptance tests… [this](https://github.com/travis-ci/travis-web/blob/f340ea6f1acc565ddbf35c350add6ad347fee5ea/tests/acceptance/dashboard/repositories-test.js) is an example of the new standard, which involves the use of `async` and `await` rather than the `andThen` blocks. Since we still have a lot of acceptance tests that use the old style, it’s not disastrous to add another, but I feel like when adding new ones, we might as well use the more modern style.

Reply:
For some reason I thought [this acceptance test](https://github.com/travis-ci/travis-web/pull/1878/files#diff-0e18780d33a062ccccacf76d2383ad51) was using the old style but I was wrong, so clearly you already know about this, sorry for the unnecessary commentary 😳

Reply:
Hey no worries! :) I'd rather have too much commentary/communication than not enough. I believe there's at least one place where I was using the old style, so it's good to know that we're trying to move away from that. And things that I do "already know", I don't always remember haha. So it's good to have reminders, and I really appreciate all the feedback I've received so far :D